qiskit/pulse/schedule.py

```diff
@@ -926,65 +478,1111 @@ def draw(self,
                                plotter=plotter,
                                axis=axis)
 
-    def __eq__(self, other: ScheduleComponent) -> bool:
-        """Test if two ScheduleComponents are equal.
+    def __len__(self) -> int:
+        """Return number of instructions in the schedule."""
+        return len(self.instructions)
 
-        Equality is checked by verifying there is an equal instruction at every time
-        in ``other`` for every instruction in this ``Schedule``.
+    def __repr__(self) -> str:
+        name = format(self._name) if self._name else ""
+        instructions = ", ".join([repr(instr) for instr in self.instructions[:50]])
+        if len(self.instructions) > 25:
+            instructions += ", ..."
+        return '{}({}, name="{}")'.format(self.__class__.__name__, instructions, name)
 
-        .. warning::
+    def __add__(self, other: 'PulseProgram') -> 'PulseProgram':
+        """Return a new schedule with ``other`` inserted within ``self`` at ``start_time``."""
+        return self.append(other)
 
-            This does not check for logical equivalency. Ie.,
+    def __lshift__(self, time: int) -> 'PulseProgram':
+        """Return a new schedule which is shifted forward by ``time``."""
+        return self.shift(time)
 
-            ```python
-            >>> (Delay(10)(DriveChannel(0)) + Delay(10)(DriveChannel(0)) ==
-                 Delay(20)(DriveChannel(0)))
-            False
-            ```
-        """
-        channels = set(self.channels)
-        other_channels = set(other.channels)
 
-        # first check channels are the same
-        if channels != other_channels:
-            return False
+class Schedule(PulseProgram):
+    """A quantum program *schedule* with exact time constraints for its instructions, operating
+    over all input signal *channels* and supporting special syntaxes for building.
 
-        # then verify same number of instructions in each
-        instructions = self.instructions
-        other_instructions = other.instructions
-        if len(instructions) != len(other_instructions):
-            return False
+    In Qiskit Pulse model [1], time overlap between pulse instructions on the same channel
+    is not allowed. This overlap constraint is immediately evaluated when a new
+    instruction is added to the ``Schedule`` object.
```

Comment:
Yes, the builder will be updated to create `ScheduleBlock`, which is much more scalable in terms of computational complexity. This will be converted into `Schedule`, but once transport layer support block representation we can directly submit block without conversion on front end. We still need schedule representation only for visualization.